### PR TITLE
Update bundled SDL2 to 2.0.14

### PR DESCRIPTION
**Pull request by couriersud**
Builds with USE_BUNDLED_LIB_SDL2=1 now work on macosx, windows (mingw) and linux.

### CODE REVIEW

File: scripts/src/osd/sdl_cfg.lua [hunk 1]
```diff
@@ -130,6 +130,10 @@ if _OPTIONS["targetos"]=="windows" then
 			"WIN32_LEAN_AND_MEAN",
 			"NOMINMAX",
 		}
+		buildoptions
+		{
+			"-Wno-undef",
+		}
```
Comment (cuavas) [inline scripts/src/osd/sdl_cfg.lua:136]: What’s this necessary for?  I can build with MinGW SDL 2.0.14 on Windows without needing this flag.  What’s different about the bundled SDL 2.0.14 that causes it to need this?  Also, it doesn’t follow the formatting, and having it here will break Visual Studio builds.

File: src/osd/sdl/sdlmain.cpp [hunk 2]
```diff
@@ -452,15 +452,16 @@ void sdl_osd_interface::init(running_machine &machine)
 			//osd_setenv(SDLENV_RENDERDRIVER, stemp, 1);
 			SDL_SetHint(SDL_HINT_RENDER_DRIVER, stemp);
 		}
+// Disabled for now. Don't try to be smarter than SDL. opengl renderdriver looks broken on windows
+#if 0 && defined(SDLMAME_WIN32)
```
Comment (cuavas) [inline src/osd/sdl/sdlmain.cpp:456]: This seems wrong.  I just tested with MAME 0243 built against MinGW SDL 2.0.14 and the OpenGL renderer works fine.

**Review by cuavas — COMMENTED** [review]
I didn’t actually look at the update to SDL itself to ensure it matches the release contents, I just scanned the list of files changed in that directory.  Given you seem to have inadvertently checked in a local revision ID in lua-zlib, please check that nothing like this has happened in SDL.

There are a few concerning things – you seem to be having issues with this bundled form of SDL 2.0.14 that don’t happen when building against MinGW SDL 2.0.14 (this is a configuration I test for every release).

**Comment by cuavas** [comment]
@mmicko you’ve worked with the bundled SDL – do you have any input for this PR?

File: scripts/src/3rdparty.lua [hunk 3]
```diff
@@ -2065,6 +2218,9 @@ end
 			"-Wno-undef",
 			"-Wno-unused-but-set-variable",
 		}
+		buildoptions {
+			"-Wno-undef",
+		}
```
[the remark this answers is not in the document]
Comment (couriersud) [reply scripts/src/3rdparty.lua:2223]: The error happens when a *.cpp file is compiled. Looks like buildoptions_c is only applied to *.c files. Additing buildoptions than fixes the error.

**Comment by mmicko** [comment]
@cuavas well first main intention for bundled SDL was enabling Visual Studio builds with SDL. Later it was added for Android and OSX. Not sure if people are at all using it on OSX since installing SDL2 official package works just fine.

As you pointed out in comment Linux support is one that was not enabled for a reason, parameters needed quite depend on environment.  Enabling video or audio driver options depends of what is available on platform and that also adds additional dependencies. Only way to properly do it for linux is to run configure/cmake script. 

@couriersud For Linux environment if you need to upgrade SDL2 version and no new one is available at your distribution I would suggest building from scripts and installing, and then pointing out makefile to use that installed version.

File: scripts/src/3rdparty.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (cuavas) [reply scripts/src/3rdparty.lua:2223]: You’ll end up with it twice for C files now, though.  You can remove it from `buildoptions_c` if you want it to apply to C, C++, Objective-C and Objective-C++, or add it to `buildoptions_cpp` if you just want it to additionally apply to C++.

But that aside, which C++ source file is it?  At a glance, I didn’t see any C++ files added in SDL itself.  It still seems to only contain C and Objective-C, so I wouldn’t expect this to actually solve anything here.

File: scripts/src/osd/sdl_cfg.lua (hunk 1 above)
Comment (couriersud) [reply scripts/src/osd/sdl_cfg.lua:136]: SDL_config_windows.h is causing issues with `#if _MSC_VER >= 1911` on line 195. I will fix the visual studio issue.

File: src/osd/sdl/sdlmain.cpp (hunk 2 above)
Comment (couriersud) [reply src/osd/sdl/sdlmain.cpp:456]: Thanks a lot for testing. Than this is an issue with VirtualBox opengl. Still, I am convinced we should remove this. It was added years ago.

File: scripts/src/3rdparty.lua [hunk 4]
```diff
@@ -2096,6 +2253,23 @@ end
 			"-Wno-unused-const-variable",
 		}
 
+	configuration { "linux-*" }
+		includedirs {
+			MAME_DIR .. "3rdparty/SDL2-override/linux/SDL2",
+		}
+		defines {
+			"USING_PREMAKE_CONFIG_H",
+		}
```
[the remark this answers is not in the document]
Comment (couriersud) [reply scripts/src/3rdparty.lua:2262]: @mmicko already commented. Yes, it depends on which development packages are installed. But at least it is possible now.

File: scripts/src/3rdparty.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (couriersud) [reply scripts/src/3rdparty.lua:2223]: One example is `3rdparty/SDL2/src/audio/wasapi/SDL_wasapi_winrt.cpp` line 21

File: 3rdparty/lua-zlib/lua_zlib.c [hunk 5]
```diff
@@ -392,7 +392,7 @@ LUALIB_API int luaopen_zlib(lua_State * const L) {
 
     SETLITERAL("_COPYRIGHT", "Copyright (c) 2009-2010 Brian Maher");
     SETLITERAL("_DESCRIPTION", "Yet another binding to the zlib library");
-    SETLITERAL("_VERSION", "lua-zlib $Id$ $Format:%d$");
+    SETLITERAL("_VERSION", "lua-zlib $Id: b619258c75f480717fad20c113c3b1a35f888ce0 $ $Format:%d$");
```
[the remark this answers is not in the document]
Comment (couriersud) [reply 3rdparty/lua-zlib/lua_zlib.c]: Fixed.

File: scripts/src/3rdparty.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (couriersud) [reply scripts/src/3rdparty.lua]: Fixed

File: scripts/src/osd/sdl_cfg.lua (hunk 1 above)
Comment (couriersud) [reply scripts/src/osd/sdl_cfg.lua]: Fixed.

**Comment by belegdol** [comment]
May I ask why the update is to 2.0.14 and not to a newer version? 2.0.22 was released a week ago.

**Comment by couriersud** [comment]
> May I ask why the update is to 2.0.14 and not to a newer version? 2.0.22 was released a week ago.

I interpreted the comment Vas made to go for 2.0.14. Happy to provide a PR for 2.0.22 once this PR has been accepted.

**Comment by cuavas** [comment]
This has been obsoleted – we no longer include SDL source.

**Pull request closed without merging**